src/main/java/org/skriptlang/skript/lang/comparator/Comparators.java

```diff
@@ -87,60 +90,86 @@ public static <T1, T2> void registerComparator(
+		if (firstType == Object.class && secondType == Object.class) {
 			throw new IllegalArgumentException("It is not possible to add a comparator between objects");
+		}
 
-		for (ComparatorInfo<?, ?> info : COMPARATORS) {
-			if (info.firstType == firstType && info.secondType == secondType) {
-				throw new SkriptAPIException(
-					"A Comparator comparing '" + firstType + "' and '" + secondType + " already exists!"
-				);
+		synchronized (COMPARATORS) {
+			for (ComparatorInfo<?, ?> info : COMPARATORS) {
+				if (info.firstType == firstType && info.secondType == secondType) {
+					throw new SkriptAPIException(
+						"A Comparator comparing '" + firstType + "' and '" + secondType + " already exists!"
+					);
+				}
 			}
+			COMPARATORS.add(new ComparatorInfo<>(firstType, secondType, comparator));
 		}
-
-		COMPARATORS.add(new ComparatorInfo<>(firstType, secondType, comparator));
 	}
 
 	/**
 	 * Compares two objects to see if a Relation exists between them.
 	 * @param first The first object for comparison.
 	 * @param second The second object for comparison.
 	 * @return The Relation between the two provided objects.
+	 * Guaranteed to be {@link Relation#NOT_EQUAL} if either parameter is null.
 	 */
 	@SuppressWarnings("unchecked")
 	public static <T1, T2> Relation compare(@Nullable T1 first, @Nullable T2 second) {
-		if (first == null || second == null)
+		assertIsDoneLoading(); // this would be checked later on too, but we want this guaranteed to fail
+
+		if (first == null || second == null) {
 			return Relation.NOT_EQUAL;
+		}
+
+		if (first == second) { // easiest check of them all!
+			return Relation.EQUAL;
+		}
 
 		Comparator<T1, T2> comparator = getComparator((Class<T1>) first.getClass(), (Class<T2>) second.getClass());
-		if (comparator == null)
+		if (comparator == null) {
 			return Relation.NOT_EQUAL;
+		}
 
 		return comparator.compare(first, second);
 	}
 
 	/**
-	 * A method for obtaining a comparator that can compare two objects of 'firstType' and 'secondType'.
+	 * A method for obtaining a Comparator that can compare two objects of 'firstType' and 'secondType'.
 	 * Please note that comparators may convert objects if necessary for comparisons.
 	 * @param firstType The first type for comparison.
 	 * @param secondType The second type for comparison.
-	 * @return A comparator capable of determine the {@link Relation} between two objects of 'firstType' and 'secondType'.
+	 * @return A Comparator capable of determine the {@link Relation} between two objects of 'firstType' and 'secondType'.
 	 * Will be null if no comparator capable of comparing two objects of 'firstType' and 'secondType' was found.
 	 */
 	@Nullable
-	@SuppressWarnings("unchecked")
 	public static <T1, T2> Comparator<T1, T2> getComparator(Class<T1> firstType, Class<T2> secondType) {
-		if (Skript.isAcceptRegistrations())
-			throw new SkriptAPIException("Comparators cannot be retrieved until Skript has finished registrations.");
```

Comment:
Shouldn't this be replaced with `assertIsDoneLoading();`?

Reply:
The method call to getComparatorInfo will immediately perform the check. I could add it but wasn't sure that it was necessary.

Reply:
Sounds not necessary.